src/webui/api/rsscontroller.cpp

```diff
@@ -139,3 +163,26 @@ void RSSController::rulesAction()
 
     setResult(jsonObj);
 }
+
+void RSSController::matchingArticlesAction()
+{
+    requireParams({"ruleName"});
+
+    const QString ruleName {params()["ruleName"]};
+    const RSS::AutoDownloadRule rule = RSS::AutoDownloader::instance()->ruleByName(ruleName);
+
+    QJsonObject jsonObj;
+    for (const QString &feedURL : rule.feedURLs()) {
+        const RSS::Feed *feed = RSS::Session::instance()->feedByURL(feedURL);
+        if (!feed) continue; // feed doesn't exist
+
+        QJsonArray matchingArticles;
+        for (const RSS::Article *article : feed->articles())
+            if (rule.matches(article->data()))
+                matchingArticles << article->title();
+        if (!matchingArticles.isEmpty())
+            jsonObj.insert(feed->name(), matchingArticles);
```

Comment:
Why do you index them by Feed name? I don't think that's a good idea. The Feed name is not immutable, so it can be changed sometime between, which will lead to conflicts.

Reply:
In the desktop UI the matching rules are ordered in sub categories by feed name. In what form should I return the data instead?

Reply:
Oops... now I understand this data is purely to "mimic" qBittorrent Desktop UI (just to show it to the user). Seems getting  "matching" articles cannot be useful for something else.
Well, leave it as is.